cmake/ZeekSpicyAnalyzerSupport.cmake

```diff
@@ -140,12 +140,20 @@ if (SPICYZ)
     add_executable(spicyz IMPORTED)
     set_property(TARGET spicyz PROPERTY IMPORTED_LOCATION "${SPICYZ}")
 
-    if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-        set(SPICYZ_FLAGS "-d")
-    else ()
+    # Allow passing -DSPICYZ_FLAGS="..."
+    if (NOT DEFINED SPICYZ_FLAGS)
         set(SPICYZ_FLAGS "")
     endif ()
 
+    # Allow passing SPICYZ_FLAGS via environment variable
+    set(SPICYZ_FLAGS "${SPICYZ_FLAGS};$ENV{SPICYZ_FLAGS}")
+
+    if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+        set(SPICYZ_FLAGS "${SPICYZ_FLAGS};-d")
+    endif ()
+
+    string(REPLACE " " ";" SPICYZ_FLAGS "${SPICYZ_FLAGS}")
```

Comment:
Just a nit: I think I'd apply this earlier just to the value of `$ENV{}`, so that `SPICYZ_FLAGS` is valid list all the time.

Reply:
Ah, hmmm 🤔... so, when `SPICYZ_FLAGS` is set via `cmake -D` then it starts life as a non-valid list (as `$ENV{SPICYZ_FLAGS}` would), so I figured: wait for all the concatenations before doing the `string(REPLACE, ...)`.

Do you know if there is another path by which `SPICYZ_FLAGS` might be set as a valid list instead?

Full disclosure: I'm still a CMake noob, so I found this list business somewhat puzzling (initially spaces were being escaped in the resulting `spicyz` command line).

Reply:
Ok, I reworked this a little bit mainly for clarity: https://github.com/zeek/spicy-plugin/compare/main...topic/robin/cmake-user-spicyz-flags.